sway-lib-std/src/bytes.sw

```diff
@@ -611,26 +621,34 @@ impl Bytes {
     /// bytes2.push(9u8);
     /// assert(bytes2.len() == 3);
     ///
-    /// let mut joined = bytes.join(bytes2);
-    /// assert(joined.len() == bytes.len() + bytes2.len());
-    /// assert(joined.capacity() == bytes.len() + bytes2.len());
+    /// let first_length = bytes.len();
+    /// let second_length = bytes2.len();
+    /// let first_cap = bytes.capacity();
+    /// let second_cap = bytes2.capacity();
+    /// bytes.append(bytes2);
+    /// assert(bytes.len() == first_length + second_length);
+    /// assert(bytes.capacity() == first_length + first_length);
     /// ```
-    pub fn join(ref mut self, other: self) -> Self {
-        let mut joined = Bytes::with_capacity(self.len + other.len);
+    pub fn append(ref mut self, ref other: self) {
+        let both_len = self.len + other.len;
+        let other_start = self.len;
```

Comment:
It may be worth returning here if `other.len` is zero?  A rare occurrence but a fair optimisation probably.

Another optimisation would be if `self.len` is zero, instead of copying the bytes, they can just be 'moved' to self, by copying the pointer and avoiding a new alloc.  This might be less rare if the user is building something starting with an empty `Bytes`.

And then adding tests for those cases too.